docroot/modules/custom/uiowa_core/uiowa_core.module

```diff
@@ -875,3 +875,54 @@ function uiowa_core_config_entity_filter_select(array &$form, $entity_id) {
   $form['#options'] = $options;
   unset($form['#size']);
 }
+
+/**
+ * Helper function to set the field label icon.
+ *
+ * @param array $variables
+ *   The field variables array.
+ * @param string $icon_classes
+ *   The Font Awesome class for the field label icon.
+ */
+function uiowa_core_set_field_label_icon(array &$variables, $icon_classes) {
+  if (!empty($icon_classes)) {
+    if (!isset($variables['field_label_icon'])) {
+      $variables['field_label_icon'] = '';
+    }
+    $variables['field_label_icon'] = $icon_classes;
+  }
+}
+
+/**
+ * Helper function to set the block/field's attributes.
+ *
+ * @param array $variables
+ *   The block/field variables array.
+ * @param string $classes
+ *   The classes to be added to the block/field's attributes.
+ */
+function uiowa_core_set_attributes(array &$variables, $classes = []) {
+  if (!empty($classes)) {
+    if (!isset($variables['attributes']['class'])) {
+      $variables['attributes']['class'] = [];
+    }
+    $variables['attributes']['class'] = [$classes];
+  }
+}
+
+/**
+ * Helper function to set the block/field's title attributes.
+ *
+ * @param array $variables
+ *   The block/field variables array.
+ * @param string $title_classes
+ *   The classes to be added to the block/field's title attributes.
+ */
+function uiowa_core_set_title_attributes(array &$variables, $title_classes = []) {
+  if (!empty($title_classes)) {
+    if (!isset($variables['title_attributes']['class'])) {
+      $variables['title_attributes']['class'] = [];
+    }
+    $variables['title_attributes']['class'] = [$title_classes];
+  }
+}
```

Comment:
Alright, so I think they need to be updated to something like 
```  
if (!empty($title_classes)) {
    if (!isset($variables['title_attributes']['class'])) {
      $variables['title_attributes']['class'] = [];
    }
    $variables['title_attributes']['class'][] = [$title_classes];
  }
```
so we don't lose the existing?

And something like this for the string?
```
$variables['field_label_icon'] = implode(' ', [$variables['field_label_icon'], $icon_classes]);
```

Reply:
Unless we want to also check that we're not adding duplicate classes if they overlap what's already been set...not sure if that gets resolved somewhere later down the line or not.

Reply:
I didn't look at this very carefully before. I think we want to think through this a little bit more. We are doing this type of thing all over the place, so the helper functions would be very useful.

It would be nice to have the class that gets passed in be able to be a string or an array. So, if you just need to add one class, you can pass `'my-class'`, but if you need to add multiple classes, you can pass an array.

Reply:
I took a pass at refactoring these a bit:

```
/**
 * Helper function to set the block/field's title attributes.
 *
 * @param array $variables
 *   The block/field variables array.
 * @param array|string $title_classes
 *   The class(es) to be added to the block/field's title attributes.
 */
function uiowa_core_set_title_attributes(array &$variables, array|string $title_classes = []): void {
  if (empty($title_classes)) {
    return;
  }
  if (!isset($variables['title_attributes']['class'])) {
    $variables['title_attributes']['class'] = [];
  }
  if (!is_array($title_classes)) {
    $title_classes = [$title_classes];
  }
  $variables['title_attributes']['class'] = array_unique(array_merge($variables['title_attributes']['class'], $title_classes));
}
```

Reply:
I think you could potentially remove the `= []` in the function argument for the classes and then remove the section related to it being empty. This would mean that PhpStorm would be able to tell us if we are not passing anything in so we can correct that. We really shouldn't be using this function if we are going to pass anything for the second argument.

Reply:
@pyrello Good point!